### DISCUSSION TITLE
Giving UI elements with numbers the ability to use math operations

### DISCUSSION THREAD

**Discussion by CookieLynx** [diesel-modding/PAYDAY-2-BeardLib#575] [Ideas]
**Is your feature request related to a problem? Please describe.**
Not related to a problem, just a QOL suggestion with being able to use math operations on UI elements that use numbers.

**Describe the solution you'd like**
Unity has a very simple system that could work well here, see screenshots below. It works with all basic math operations (+, -, /, *) it also works with order of operations but that's not required.

![image](https://user-images.githubusercontent.com/52944061/141660841-f480e1bd-fdf3-42bb-b36f-a9943db3a93c.png)
Original values

![image](https://user-images.githubusercontent.com/52944061/141660857-26f34673-7f5d-484e-9513-f7396f14dd36.png)
Adding 30

![image](https://user-images.githubusercontent.com/52944061/141660866-79a071bb-57e6-459b-8a18-bd0dbdbfde84.png)
After pressing enter or clicking off of value

This would be very useful for doing simple math in game quickly without having to alt tab out to a calculator.

**Comment by segabl** [comment]
This looks useful and not too hard to implement, I may give this a try

**Reply by segabl** [reply]
you can try out this feature on the https://github.com/simon-wh/PAYDAY-2-BeardLib/tree/expression-eval branch
